src/EditorFeatures/CSharpTest/CodeActions/InlineTemporary/InlineTemporaryTests.cs

```diff
@@ -3950,5 +3951,102 @@ static void N()
 
             await TestAsync(initial, expected, compareTokens: false);
         }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineTemporary), Trait(Traits.Feature, Test.Utilities.Traits.Features.Tuples)]
+        public async Task Tuples_Disabled()
+        {
+            var code = @"
+using System;
+class C
+{
+    public void M()
+    {
+        (int, string) [||]x = (1, ""hello"");
+        x.ToString();
+    }
+}";
+
+            await TestMissingAsync(code, parseOptions: TestOptions.Regular.WithTuplesFeature());
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineTemporary), Trait(Traits.Feature, Test.Utilities.Traits.Features.Tuples)]
+        public async Task Tuples()
+        {
+            var code = @"
+using System;
+class C
+{
+    public void M()
+    {
+        (int, string) [||]x = (1, ""hello"");
+        x.ToString();
+    }
+}";
+
+            var expected = @"
+using System;
+class C
+{
+    public void M()
+    {
+        ((1, ""hello"")).ToString();
+    }
+}";
+
+            await TestAsync(code, expected, index: 0, compareTokens: false, parseOptions: TestOptions.Regular.WithTuplesFeature());
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineTemporary), Trait(Traits.Feature, Test.Utilities.Traits.Features.Tuples)]
+        public async Task TuplesWithNames()
+        {
+            var code = @"
+using System;
+class C
+{
+    public void M()
+    {
+        (int a, string b) [||]x = (a: 1, b: ""hello"");
+        x.ToString();
+    }
+}";
+
+            var expected = @"
+using System;
+class C
+{
+    public void M()
+    {
+        ((a: 1, b: ""hello"")).ToString();
+    }
+}";
+
+            await TestAsync(code, expected, index: 0, compareTokens: false, parseOptions: TestOptions.Regular.WithTuplesFeature());
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineTemporary), Trait(Traits.Feature, Test.Utilities.Traits.Features.Tuples)]
+        public async Task TuplesWithDifferentNames()
+        {
+            // Note: this refactoring is less than ideal
+            var code = @"
+class C
+{
+    public void M()
+    {
+        (int a, string b) [||]x = (c: 1, d: ""hello"");
+        x.a.ToString();
+    }
+}";
+
+            var expected = @"
+class C
+{
+    public void M()
+    {
+        (((int a, string b))((c: (int)1, d: (string)""hello""))).a.ToString();
```

Comment:
That'll take work in the simplifier. Feel free to file an issue if you haven't already.